src/AppButton.js

```diff
@@ -71,6 +71,8 @@ export default function AppButton({ appInfo, content_type, entry, window }) {
       save: true,
     });
     if (close_on_success && success) {
+      console.log("Appbutton2 entry.get_text: "+ entry.get_text());
+      CommonSettings.openedWindows.delete(entry.get_text());
```

Comment:
Approach should be to detect opened windows and act on this information. For example window could be killed by third party program, then this logic would not work.